cumulusci/tasks/apex/anon.py

```diff
@@ -1,35 +1,110 @@
+        },
+        "namespaced": {
+            "description": (
+                "If True, the tokens %%%NAMESPACED_RT%%% and %%%namespaced%%% "
+                "will get replaced with the namespace prefix for Record Types."
+            ),
+            "required": False,
+        },
+    }
+
+    def _validate_options(self):
+        super(AnonymousApexTask, self)._validate_options()
 
-    task_options = {"apex": {"description": "The apex to run.", "required": True}}
+        if not self.options.get('path') and not self.options.get('apex'):
+            raise TaskOptionsError('You must specify either the `path` or `apex` option.')
 
     def _run_task(self):
-        self.logger.info("Executing Anonymous Apex")
+        apex = ''
+        apex_path = self.options.get('path')
+        if apex_path:
+            if not in_directory(apex_path, self.project_config.repo_root):
+                raise TaskOptionsError(
+                    'Please specify a path inside your project repository. '
+                    'You specified: {}'.format(apex_path)
+                )
+            self.logger.info("Executing anonymous Apex from {}".format(apex_path))
+            try:
+                with open(apex_path, 'r') as f:
+                    apex = f.read()
+            except IOError:
+                raise TaskOptionsError("Could not find or read file: {}".format(apex_path))
+        else:
+            self.logger.info('Executing anonymous Apex')
+
+        apex_string = self.options.get('apex')
+        if apex_string:
+            apex = apex + '\n' + apex_string
+
+        apex = self._prepare_apex(apex)
         result = self.tooling._call_salesforce(
             method="GET",
             url="{}executeAnonymous".format(self.tooling.base_url),
-            params={"anonymousBody": self.options["apex"]},
+            params={"anonymousBody": apex},
         )
-        if result.status_code != 200:
-            raise SalesforceGeneralError(url, path, result.status_code, result.content)
+        self._check_result(result)
+
+        self.logger.info("Anonymous Apex Success")
+
+    def _prepare_apex(self, apex):
+        # Replace the email string only if it exists
+        admin_email = self.project_config.dev_config__admin_email
+        if admin_email and EMAIL_TOKEN in apex:
+            self.logger.info("Replacing {} with {}".format(EMAIL_TOKEN, admin_email))
+            apex = apex.replace(EMAIL_TOKEN, admin_email)
+
+        # Process namespace tokens
+        managed = self.options.get('managed') or False
+        namespaced = self.options.get('namespaced') or False
+        namespace = self.project_config.project__package__namespace
+        namespace_prefix = ''
+        record_type_prefix = ''
+        if managed or namespaced:
+            namespace_prefix = namespace + '__'
+        if namespaced:
+            record_type_prefix = namespace + '.'
+        apex = apex.replace('%%%NAMESPACE%%%', namespace_prefix)
```

Comment:
run `black`

Reply:
Okay, actually got it set up to format on save now.